### PR TITLE
SAK-47640 Rubrics: Min & Max Score for the Rubric & each Criterion + total weight warning

### PR DESCRIPTION
**Pull request by V100X**
Jira: [SAK-47640](https://sakaiproject.atlassian.net/browse/SAK-47640)

weightedRubricError_after

### CODE REVIEW

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js [hunk 1]
```diff
@@ -390,6 +397,27 @@ export class SakaiRubric extends RubricsElement {
...
+  getMinPoints(criterions) {
+    return this.getPoints(criterions, Math.min);
+  }
+
+  getPoints(criterions, minOrMax) {
+    let totalPoints = 0;
+    criterions
+      .filter( (criterion) => criterion.ratings.length > 0)
+      .forEach( (criterion) => {
+        totalPoints += minOrMax(...criterion.ratings.map( (rating) => {
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:415]: You should extract this into a function as you are using it in a few places.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js [hunk 2]
```diff
@@ -390,6 +397,27 @@ export class SakaiRubric extends RubricsElement {
...
+  getMaxPoints(criterions) {
+    return this.getPoints(criterions, Math.max);
+  }
+
+  getMinPoints(criterions) {
+    return this.getPoints(criterions, Math.min);
+  }
+
+  getPoints(criterions, minOrMax) {
+    let totalPoints = 0;
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:411]: I know I really annoy people asking this, but can you please leave a blank line before the method body? The rest of the rubrics code does this and I think it makes the code more readable.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js [hunk 3]
```diff
@@ -390,6 +397,27 @@ export class SakaiRubric extends RubricsElement {
...
+    criterions
+      .filter( (criterion) => criterion.ratings.length > 0)
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:413]: I wouldn't bother with this filter, personally. This causes a pass over the whole array and I think you may as well just let the map do its thing.

**Review by adrianfish — CHANGES_REQUESTED** [review]
This is a nice PR, thanks. Just a few small things.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js (hunk 1 above)
Comment (V100X) [reply webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:415]: Hi, where would you place the new function?

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:415]: I'd put it in sakai-rubrics-utils.js

Comment (V100X) [reply webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:415]: I'm having trouble to successfully call the method once located on `sakai-rubrics-utils.js`.
Is there any special way for this case?

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:415]: You need to import it at the top of sakai-rubric.js, es6 style. Something like:

import { calculatePoints } from "./sakai-rubric-utils.js";

you also need to export it in sakai-rubric-utils.js

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:415]: maybe call the function calculateCriteriaPoints, something like that?

**Comment by V100X** [comment]
Okay, should be ready now 👍🏽

**Review by adrianfish — APPROVED** [review]
Looks good. Thanks for the work :)

**Pull request merged**